### DISCUSSION TITLE
How to setflags() for pybind11 array ?

### DISCUSSION THREAD

**Discussion by jiapei100** [pybind/pybind11#3181] [General]
From https://github.com/pybind/pybind11/blob/master/include/pybind11/numpy.h, we can see there is a function 

```python
 /// Return the NumPy array flags
    int flags() const {
        return detail::array_proxy(m_ptr)->flags;
    }
```
which is returning array's **flags**. Just wonder if there is a way to **setflags()** for **pybind11 array**?

Cheers

**Comment by Skylion007** [comment]
The Flags are template param of array_t object in detail/numpy.h and are defined a construction of the array_t. If you want to change the flags you will need to construct a new array_t wrapper object with the flags you desire: https://github.com/pybind/pybind11/blob/0be2ea065eeaf2983964ecb792d30abbcedcdc9d/include/pybind11/numpy.h#L846

**Comment by Skylion007** [comment]
@jiapei100 Our Eigen bindings do this with the writable-flag and the array_proxy class btw.
